Http/Http.cs

```diff
@@ -0,0 +1,352 @@
+		/// </summary>
+		/// <param name="uri">The URI to which the data will be sent.</param>
+		/// <param name="bodyData">The data to transmit to the remote server.
+		/// The string will be converted to raw bytes via &lt;a href="http:msdn.microsoft.comen-uslibrarysystem.text.encoding.utf8"&gt;System.Text.Encoding.UTF8&lt;a&gt;.</param>
+		/// <returns>
+		///   <para>A HttpRequest configured to transmit bodyData to uri via HTTP PUT.</para>
+		/// </returns>
+		public static HttpRequest Put(string uri, string bodyData)
+		{
+			return new HttpRequest(UnityWebRequest.Put(uri, bodyData));
+		}
+
+		/// <summary>
+		///   <para>Creates a HttpRequest configured for HTTP DELETE.</para>
+		/// </summary>
+		/// <param name="uri">The URI to which a DELETE request should be sent.</param>
+		/// <returns>
+		///   <para>A HttpRequest configured to send an HTTP DELETE request.</para>
+		/// </returns>
+		public static HttpRequest Delete(string uri)
+		{
+			return new HttpRequest(UnityWebRequest.Delete(uri));
+		}
+
+		/// <summary>
+		///   <para>Creates a HttpRequest configured to send a HTTP HEAD request.</para>
+		/// </summary>
+		/// <param name="uri">The URI to which to send a HTTP HEAD request.</param>
+		/// <returns>
+		///   <para>A HttpRequest configured to transmit a HTTP HEAD request.</para>
+		/// </returns>
+		public static HttpRequest Head(string uri)
+		{
+			return new HttpRequest(UnityWebRequest.Head(uri));
+		}
+
+		#endregion
+
+		#region None Static Send Methods
+
+		public void Send(HttpRequest request, Action<HttpResponse> onSuccess = null, Action<HttpResponse> onError = null)
+		{
+			StartCoroutine(SendCoroutine(request, onSuccess, onError));
+		}
+
+		public void Send(UnityWebRequest unityWebRequest, Action<UnityWebRequest> onSuccess = null,
+			Action<UnityWebRequest> onError = null, Action<UnityWebRequest> onNetworkError = null)
+		{
+			StartCoroutine(SendCoroutine(unityWebRequest, onSuccess, onError, onNetworkError));
+		}
+
+		public void Send(UnityWebRequest unityWebRequest, Action onSuccess = null,
+			Action onError = null, Action onNetworkError = null)
+		{
+			StartCoroutine(SendCoroutine(unityWebRequest, onSuccess, onError, onNetworkError));
+		}
+
+		public void Send(UnityWebRequest unityWebRequest, Action<HttpResponse> onSuccess = null,
+			Action<HttpResponse> onError = null)
+		{
+			StartCoroutine(SendCoroutine(unityWebRequest, onSuccess, onError));
+		}
+
+		#endregion
+
+		#region Send HttpRequest methods
+
+		private static IEnumerator SendCoroutine(HttpRequest request, Action<HttpResponse> onSuccess,
+			Action<HttpResponse> onError)
+		{
+			var unityWebRequest = request.UnityWebRequest;
+			yield return unityWebRequest.SendWebRequest();
+			var response = new HttpResponse(unityWebRequest);
+
+			if (onError != null && (unityWebRequest.isNetworkError || unityWebRequest.isHttpError))
+			{
+				onError.Invoke(response);
+			}
+			else if (onSuccess != null)
```

Comment:
This could still get called if `(unityWebRequest.isNetworkError || unityWebRequest.isHttpError)` is true, but the `onError` action is null. You should separate the action null checks from the error logic